cli/src/main.rs

```diff
@@ -0,0 +1,261 @@
+        map: String,
+    },
+    /// Import a JSON scenario in the
+    /// https://a-b-street.github.io/docs/tech/dev/formats/scenarios.html format
+    ImportScenario {
+        /// The path to a JSON scenario file
+        #[structopt(long)]
+        input: String,
+        /// The path to a map matching the scenario data
+        #[structopt(long)]
+        map: String,
+        /// Problems occur when a position is within the map boundary, but not close enough to
+        /// buildings. Skip people with problematic positions if true, abort otherwise.
+        #[structopt(long)]
+        skip_problems: bool,
+    },
+    /// Transform a JSON map that's been manually edited into the binary format suitable for
+    /// simulation.
+    ImportJSONMap {
+        /// The path to a JSON map file to import
+        #[structopt(long)]
+        input: String,
+        /// The path to write
+        #[structopt(long)]
+        output: String,
+    },
+    /// Removes nonessential parts of a Map, for the bike network tool.
+    MinifyMap {
+        /// The path to a map to shrink
+        #[structopt()]
+        map: String,
+    },
+    /// Procedurally generates houses along empty residential roads of a map
+    GenerateHouses {
+        /// The path to a map to generate houses for
+        #[structopt(long)]
+        map: String,
+        /// If the tool doesn't generate at least this many houses, then fail. This can be used to
+        /// autodetect if a map probably already has most houses tagged in OSM.
+        #[structopt(long)]
+        num_required: usize,
+        /// A seed for generating random numbers
+        #[structopt(long, default_value = "42")]
+        rng_seed: u64,
+        /// The GeoJSON file to write
+        #[structopt(long)]
+        output: String,
+    },
+    /// Prints the osm.pbf file from download.geofabrik.de that covers a given boundary.
+    ///
+    /// This is a useful tool when importing a new map, if you don't already know which geofabrik
+    /// file you should use as your OSM input.
+    PickGeofabrik {
+        /// The path to an [osmosis polygon boundary
+        /// file](https://wiki.openstreetmap.org/wiki/Osmosis/Polygon_Filter_File_Format)
+        #[structopt()]
+        input: String,
+    },
+    /// Imports a one-shot A/B Street map in a single command.
+    OneStepImport {
+        /// The path to a GeoJSON file with a boundary
+        #[structopt()]
+        geojson_path: String,
+        /// Do people drive on the left side of the road in this map?
+        #[structopt(long)]
+        drive_on_left: bool,
+        /// Use Geofabrik to grab OSM input if true, or Overpass if false. Overpass is faster.
+        #[structopt(long)]
+        use_geofabrik: bool,
+    },
+    /// Runs the main A/B Street importer, which manages maps and scenarios for many cities.
+    Import {
+        /// See the importer's source code for the defined flags. You should first pass a bare "--"
```

Comment:
I gave up on defining all of the flags here, for now. :\  I need to learn more structopt to understand how to do enumerations of flags -- like if you pass `--oneshot`, only then does `--oneshot_drive_on_left` become a valid flag. Possibly the proper way to do this is simply split out two subcommands, for a "normal" import and for a one-shot.